lib/pbench/server/database/models/datasets.py

```diff
@@ -436,11 +435,11 @@ class Dataset(Database.Base):
     # Dataset name
     name = Column(String(255), unique=False, nullable=False)
 
-    # ID of the owning user
-    owner_id = Column(Integer, ForeignKey("users.id"), nullable=False)
+    # Keycloak ID of the owning user
+    owner_id = Column(String(255), nullable=True)
```

Comment:
Why is the dataset owner now `nullable`? We might conceivably have a "stale" owner if a Keycloak user is deleted, but we already know we'll need some admin API to reassign ownership.

I'm not entirely sure this is a big deal, either, as long as it's public; but I'd prefer not to drop that constraint unless there's a really good reason.

Reply:
To reiterate: I don't particularly like (nor see any need for) "unowned" datasets. We may support changing owners, or possibly even assigning ownership to a group, but they're always owned by *someone*, so we want `nullable=False`.